ambry-replication/src/main/java/com/github/ambry/replication/ReplicaThread.java

```diff
@@ -390,9 +411,144 @@ public void replicate() {
     }
   }
 
-  //TODO implement continuous replication inside this
+  /**
+   * Polls the give remote replica groups for requests, submits these requests to networkClient,
+   * gets responses from networkClient and submits these to corresponding remote replica groups
+   *
+   * @param remoteReplicaGroups remoteReplica groups
+   * @param correlationIdToReplicaGroup map of correlation id to remote replica group
+   * @param correlationIdToRequestInfo map of correlation id to request info
+   */
+  private void pollRequestsAndSubmitResponses(List<RemoteReplicaGroup> remoteReplicaGroups,
+      Map<Integer, RemoteReplicaGroup> correlationIdToReplicaGroup,
+      Map<Integer, RequestInfo> correlationIdToRequestInfo) {
+
+    List<RequestInfo> requestInfos =
+        pollRemoteReplicaGroups(remoteReplicaGroups, correlationIdToRequestInfo, correlationIdToReplicaGroup);
+    logger.trace("Thread name: {} Requests generated", threadName);
+
+    // generate responses for requests that are timed out
+    List<ResponseInfo> responseInfosForTimedOutRequests =
+        filterTimedOutRequests(correlationIdToRequestInfo, correlationIdToReplicaGroup);
+    logger.trace("Thread name: {} Filtered timed out Requests", threadName);
+
+    // mark timed out requests to be dropped by network client
+    Set<Integer> requestsToDrop = responseInfosForTimedOutRequests.stream()
```

Comment:
@mudit-saxena we have config here too drop request on timeout from replication side too.